tests/python/gaia-ui-tests/gaiatest/tests/functional/messages/test_sms_notification.py

```diff
@@ -18,12 +18,19 @@ def test_sms_notification(self):
 
         system = System(self.marionette)
 
+        self.assertEqual(0, len(self.data_layer.get_all_sms()))
```

Comment:
Usually assertion are not done at the beginning of a test, as we assume the device is in a clean state. Also this assertion can fail (because an SMS just got in after the clean up, for instance), I'd recommend to set up an init variable like 
``` python
initial_number_of_message = len(self.data_layer.get_all_sms())
``` 
and then use this value.